File: MCP服务器/finance/QuantConnect_mcp-server_3a656b0_python/tests/algorithms/OrderProperties.cs
#region imports
    using System;
    using QuantConnect;
    using QuantConnect.Algorithm;
    using QuantConnect.Data;
    using QuantConnect.Orders;
    using QuantConnect.Securities.Equity;
#endregion

public class OrderPropertiesTestAlgorithm : QCAlgorithm
{
    private Equity _equity;

    public override void Initialize()
    {
        SetStartDate(2024, 1, 8);
        SetEndDate(2024, 2, 1);
        SetCash(100000);
        _equity = AddEquity("SPY", Resolution.Hour, dataNormalizationMode: DataNormalizationMode.Raw);
    }

    public override void OnData(Slice data)
    {
        if (Portfolio.Invested)
        {
            return;
        }
        string tag = "some tag";
        int quantity = 1;
        decimal limitPrice = _equity.Price + 10;

        // Test the GoodTilCanceled time in force
        LimitOrder(_equity.Symbol, quantity, limitPrice, tag, new OrderProperties {TimeInForce = TimeInForce.GoodTilCanceled});

        // Test the Day time in force
        LimitOrder(_equity.Symbol, quantity, limitPrice, tag, new OrderProperties {TimeInForce = TimeInForce.Day});

        // Test the GoodTilDate time in force
        LimitOrder(_equity.Symbol, quantity, limitPrice, tag, new OrderProperties {TimeInForce = TimeInForce.GoodTilDate(new DateTime(2025, 1, 1))});
    }
}

File: MCP服务器/finance/QuantConnect_mcp-server_3a656b0_python/tests/algorithms/order_properties.py
# region imports
from AlgorithmImports import *
# endregion


class OrderPropertiesTestAlgorithm(QCAlgorithm):

    def initialize(self):
        self.set_start_date(2024, 1, 8)
        self.set_end_date(2024, 2, 1)
        self.set_cash(100_000)
        self._equity = self.add_equity(
            'SPY', Resolution.HOUR, 
            data_normalization_mode=DataNormalizationMode.RAW
        )
        
    def on_data(self, data: Slice):
        if self.portfolio.invested:
            return
        tag = 'some tag'
        quantity = 1
        limit_price = self._equity.price + 10
        # Test the GOOD_TIL_CANCELED time in force.
        order_properties = OrderProperties()
        order_properties.time_in_force = TimeInForce.GOOD_TIL_CANCELED
        self.limit_order(
            self._equity.symbol, quantity, limit_price, tag, order_properties
        )

        # Test the Day time in force.
        order_properties = OrderProperties()
        order_properties.time_in_force = TimeInForce.DAY
        self.limit_order(
            self._equity.symbol, quantity, limit_price, tag, order_properties
        )

        # Test the good_til_date time in force.
        order_properties = OrderProperties()
        order_properties.time_in_force = TimeInForce.good_til_date(
            datetime(2025, 1, 1)
        )
        self.limit_order(
            self._equity.symbol, quantity, limit_price, tag, order_properties
        )
